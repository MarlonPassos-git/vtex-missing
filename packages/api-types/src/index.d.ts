export { SearchApi } from './search_api';
