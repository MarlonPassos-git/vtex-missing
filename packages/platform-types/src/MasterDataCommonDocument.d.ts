export interface CommonMasterDataDocument {
  id: string;
  accountId: string;
  accountName: string;
  dataEntityId: string;
  createdBy: string;
  createdIn: string;
  updatedBy: string | null;
  updatedIn: string | null;
  lastInteractionBy: string;
  lastInteractionIn: string;
  followers: unknown[];
  tags: unknown[];
  auto_filter: unknown | null;
}

export interface CommonMasterDataV1Document extends CommonMasterDataDocument {

}

export interface CommonMasterDataV2Document extends CommonMasterDataDocument {
  schemas: string[]
  createdBy_USER: MasterDataUser
  lastInteractionBy_USER: MasterDataUser
  dataInstanceId: string
  _sort: number[]
}

type MasterDataUser = {
  Id: string,
  Login: string
  Name: null
}


