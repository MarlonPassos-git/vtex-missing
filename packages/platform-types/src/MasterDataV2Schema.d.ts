export type MasterDataV2Schema = {
  title?: string
  /**
   * Use the property v-cache to disable default caching.
   *
   * @see https://developers.vtex.com/docs/guides/starting-to-work-on-master-data-with-json-schema#disabling-default-caching
   */
  'v-cache'?: boolean
  /**
   * Use the property v-default-fields to configure which fields will return without indication in the fields query string.
   *
   * @see https://developers.vtex.com/docs/guides/starting-to-work-on-master-data-with-json-schema#default-fields
   */
  'v-default-fields'?: string[]
  /**
   * Use the property v-indexed to set up indexed fields. You must add the field to the properties to generate the indexer configuration with the right type.
   *
   * @see https://developers.vtex.com/docs/guides/starting-to-work-on-master-data-with-json-schema#indexing-fields
   */
  'v-indexed'?: string[]
  required?: string[]
  /**
   * Use the property v-security to set up which fields are public (request without user authentication).
   *
   * @see https://developers.vtex.com/docs/guides/starting-to-work-on-master-data-with-json-schema#enabling-public-fields
   */
  'v-security'?: VSecurity
  properties: Record<string, Property>
  'v-immediate-indexing'?: boolean
  'v-triggers'?: VTrigger[]
  /**
   * Use the property v-canonicalto to set up another JSON Schema in the same data entity to inheritance.
   *
   * @see https://developers.vtex.com/docs/guides/starting-to-work-on-master-data-with-json-schema#inheritance-of-schemas
   */
  'v-canonicalto'?: string
}

type VTrigger = {
  name: string
  active: boolean
  condition?: string
  action: {
    type: "save"
    dataEntity: string
    json: Record<string, string>
  }
}

type VSecurity = {
  allowGetAll?: boolean
  publicRead?: string[]
  publicWrite?: string[]
  publicJsonSchema?: boolean
  publicFilter?: string[]
}

type PropertyType = 'string' | 'number' | 'boolean' | 'array' | 'object' | 'null' | "integer"
type PropertyFormat = "date-time"

type Property = {
  /**
   * If you want to this property should to not be mandatory use an array with null and your type like this: ['string', 'null']
   */
  type: PropertyType | PropertyType[]
  unique?: boolean
  title?: string
  properties?: Record<string, Property>
  'v-indexed'?: string[]
  format?: PropertyFormat
  maxLength?: number
  enum?: string[]
  required?: boolean
  items?: Property
}

