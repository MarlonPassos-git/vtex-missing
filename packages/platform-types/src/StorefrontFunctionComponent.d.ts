import { FunctionComponent } from 'react';

type GenericObject = Record<string, unknown>;

/**
 * VTEX IO Storefront Function Component
 */
export interface StorefrontFunctionComponent<Props = GenericObject>
  extends FunctionComponent<Props> {
  getSchema?: (props: Props) => Schema;
  schema?: Schema;
  defaultProps?: Props;
}

interface Schema {
  title: string;
  type?: Extract<SchemaType, 'object'>;
  properties?: SchemaProperties;
  dependencies?: SchemaDependencies;
}

export type SchemaType = 'array' | 'object' | 'string' | 'boolean' | 'number';

export type UiWidget =
  | 'image-uploader'
  | 'textarea'
  | 'datetime'
  | 'select'
  | 'color';

export interface SchemaDependencies {
  [key: string]: {
    oneOf: {
      properties: {
        [key: string]: Partial<SchemaItemProperty>;
      };
    }[];
  };
}

export interface SchemaProperties {
  __editorItemTitle?: {
    default: string;
    title: string;
    type: Extract<SchemaType, 'string'>;
  };
  [key: string]: SchemaItemProperty;
}

export interface SchemaItemProperty {
  type: SchemaType;
  enumNames?: string[]; //esse funciona junto com o enum
  enum?: string[];
  default?: unknown;
  title?: string;
  description?: string;
  properties?: SchemaProperties;
  format?: 'date-time';
  widget?: {
    'ui:widget'?: UiWidget;
  };
  items?: {
    type: SchemaType;
    title?: string;
    properties?: SchemaProperties;
    default?: unknown;
  };
}

