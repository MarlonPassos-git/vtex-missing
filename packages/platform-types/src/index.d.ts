export { MasterDataV2Schema } from "./MasterDataV2Schema"
export { StorefrontFunctionComponent } from "./StorefrontFunctionComponent"