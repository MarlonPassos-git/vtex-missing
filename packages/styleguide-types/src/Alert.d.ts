import type { FC } from 'react';

/**
 * @see [oficial documentation](https://styleguide.vtex.com/#/Components/Notification/Alert)
 */
type Alert = FC<Props>;

type Props = {
  /**
   * Content of the alert
   *
   * @required
   */
  children: React.ReactNode;

  /**
   * Style of the alert
   *
   * @required
   * @enum {string} - One of: 'success', 'error', 'warning'
   */
  type: 'success' | 'error' | 'warning';

  /**
   * If this object is defined, an action button will appear on the right side of the alert.
   */
  action?: {
    /**
     * Required function to be called when the action button is clicked.
     *
     * @required
     */
    onClick: () => void;

    /**
     * Required label for the action button.
     *
     * @required
     */
    label: React.ReactNode;
  };

  /**
   * Time in ms to auto close the alert
   */
  autoClose?: number;

  /**
   * Defines the title used for hover and accessibility feedback
   */
  closeIconLabel?: string;

  /**
   * Set focus to the first focusable element inside the alert, which should be the "action" when available or the "close" button
   */
  focusOnOpen?: boolean;

  /**
   * If this function is defined, a close icon will appear, and this function will be called when the alert is closed.
   */
  onClose?: () => void;
};
