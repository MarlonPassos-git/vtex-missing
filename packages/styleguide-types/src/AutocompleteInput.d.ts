import type { FC, ReactNode } from 'react';

/**
 * @see [oficial documentation](https://styleguide.vtex.com/#/Components/Forms/AutocompleteInput)
 */
type AutocompleteInput = FC<Props>;

type AutocompleteInputProps = {
  input: Input;
  options: Options;
};

type Input = {
  onClear: () => void;
  onSearch: (...args: unknown[]) => void;
  onChange: (data?: unknown) => void;
  value?: string;
  disabled?: boolean;
  error?: boolean;
  placeholder?: string;
  errorMessage?: string;
  prefix?: ReactNode;
  suffix?: ReactNode;
};

type Options = {
  loading?: boolean;
  renderOption?: (option?: unknown) => ReactNode;
  value?: Value[];
  icon?: ReactNode;
  onSelect: (data?: Value) => void;
  lastSearched?: unknown;
  size?: Size;
  customMessage?: ReactNode;
};

type Value = {
  value: TValue;
  label: string;
};

type Size = 'small' | 'regular' | 'large';