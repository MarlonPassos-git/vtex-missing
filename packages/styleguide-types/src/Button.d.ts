import type { FC, PropsWithChildren } from 'react';

/**
 * @see [oficial documentation](https://styleguide.vtex.com/#/Components/Forms/Button)
 */
type Button = FC<PropsWithChildren<Props>>;

/**
 * @todo add description of props
 */
type Props = {
  /**
   * @deprecated
   * @default false
   */
  icon?: boolean;
  autoComplete?: string;
  /**
   * @default false
   */
  autoFocus?: boolean;
  /**
   * @default false
   */
  block?: boolean;
  /**
   * @default false
   */
  disabled?: boolean;
  download?: boolean;
  href?: string;
  id?: string;
  /**
   * @default false
   */
  isActiveOfGroup?: boolean;
  /**
   * @default false
   */
  isFirstOfGroup?: boolean;
  /**
   * @default false
   */
  isGrouped?: boolean;
  /**
   * @default false
   */
  isLastOfGroup?: boolean;
  /**
   * @default false
   */
  isLoading?: boolean;
  name?: string;
  noWrap?: boolean;
  onClick?: () => void;
  onMouseEnter?: () => void;
  onMouseLeave?: () => void;
  onMouseOver?: () => void;
  onMouseOut?: () => void;
  onMouseUp?: () => void;
  onMouseDown?: () => void;
  onFocus?: () => void;
  onBlur?: () => void;
  onTouchEnd?: () => void;
  onTouchMove?: () => void;
  onTouchStart?: () => void;
  referrerPolicy?: string;
  rel?: string;
  size?: Size;
  /**
   * @default 0
   */
  tabIndex?: number;
  target?: string;
  testId?: string;
  /**
   * @default 'button'
   */
  type?: string;
  value?: string;
  /**
   * @default 'primary'
   */
  variation?: Variation;
};

type Size = 'small' | 'regular' | 'large';

type Variation =
  | 'primary'
  | 'secondary'
  | 'tertiary'
  | 'inverted-tertiary'
  | 'danger'
  | 'danger-tertiary';
