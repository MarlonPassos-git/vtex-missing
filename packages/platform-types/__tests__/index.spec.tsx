import React from 'react';
import { testType } from './test-utils';
import { StorefrontFunctionComponent, MasterDataV2Schema } from "../src"

// @dts-jest:group Alert
{
  // @dts-jest:pass:snap
  testType<StorefrontFunctionComponent>(() => <div>children</div>);
}

// @dts-jest:group Dropdown
{
  // @dts-jest:pass:snap
  testType<MasterDataV2Schema>({
    title: "Basic information of SKU",
    properties: {
      "name": { "type": "string" }
    },
    required: ["name"]
  });
}

