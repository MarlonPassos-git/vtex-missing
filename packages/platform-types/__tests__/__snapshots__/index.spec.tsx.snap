// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert testType<StorefrontFunctionComponent>(() => <div>children</div>) (type) should match snapshot 1`] = `"StorefrontFunctionComponent<Record<string, unknown>>"`;

exports[`Dropdown testType<MasterDataV2Schema>({
      title: "Basic information of SKU",
      properties: {
        "name": { "type": "string" }
      },
      required: ["name"]
    }) (type) should match snapshot 1`] = `"MasterDataV2Schema"`;
